pkg/beholder/client.go

```diff
@@ -345,7 +354,11 @@ func newMeterProvider(config Config, resource *sdkresource.Resource, creds crede
 	opts := []otlpmetricgrpc.Option{
 		otlpmetricgrpc.WithTLSCredentials(creds),
 		otlpmetricgrpc.WithEndpoint(config.OtelExporterGRPCEndpoint),
-		otlpmetricgrpc.WithHeaders(config.AuthHeaders),
+	}
+	if config.AuthHeaderProvider != nil {
+		opts = append(opts, otlpmetricgrpc.WithDialOption(grpc.WithPerRPCCredentials(config.AuthHeaderProvider.Credentials())))
```

Comment:
It seems there should be a relationship between otlpmetricgrpc.WithTLSCredentials (line 355) and 
config.AuthHeaderProvider.Credentials().RequireTransportSecurity

If TLS is enabled AuthHeaderProvider.Credentials().RequireTransportSecurity()should always return true. 
Im wondering whats the good way to handle that?

Reply:
Lets test what happens when RequireTransportSecurity and InsecureConnection mismatch.